allennlp/models/decomposable_attention.py

```diff
@@ -0,0 +1,273 @@
+from typing import Dict, Optional
+
+import torch
+
+from allennlp.common import Params, constants
+from allennlp.common.checks import ConfigurationError
+from allennlp.common.tensor import get_text_field_mask, last_dim_softmax, weighted_sum
+from allennlp.common.tensor import arrays_to_variables
+from allennlp.data import Instance, Vocabulary
+from allennlp.data.fields import TextField
+from allennlp.models.model import Model
+from allennlp.modules import FeedForward, MatrixAttention
+from allennlp.modules import Seq2SeqEncoder, SimilarityFunction, TimeDistributed, TextFieldEmbedder
+
+
+@Model.register("decomposable_attention")
+class DecomposableAttention(Model):
+    """
+    This ``Model`` implements the Decomposable Attention model described in "A Decomposable
+    Attention Model for Natural Language Inference", by Parikh et al., 2016, with some optional
+    enhancements before the decomposable attention actually happens.  Parikh's original model
+    allowed for computing an "intra-sentence" attention before doing the decomposable entailment
+    step.  We generalize this to any :class:`Seq2SeqEncaoder` that can be applied to the premise
+    and/or the hypothesis before computing entailment.
+
+    The basic outline of this model is to get an embedded representation of each word in the
+    premise and hypothesis, align words between the two, compare the aligned phrases, and make a
+    final entailment decision based on this aggregated comparison.  Each step in this process uses
+    a feedforward network to modify the representation.
+
+    Parameters
+    ----------
+    vocab : ``Vocabulary``
+    text_field_embedder : ``TextFieldEmbedder``
+        Used to embed the ``premise`` and ``hypothesis`` ``TextFields`` we get as input to the
+        model.
+    attend_feedforward : ``FeedForward``
+        This feedforward network is applied to the encoded sentence representations before the
+        similarity matrix is computed between words in the premise and words in the hypothesis.
+    similarity_function : ``SimilarityFunction``
+        This is the similarity function used when computing the similarity matrix between words in
+        the premise and words in the hypothesis.
+    compare_feedforward : ``FeedForward``
+        This feedforward network is applied to the aligned premise and hypothesis representations,
+        individually.
+    aggregate_feedforward : ``FeedForward``
+        This final feedforward network is applied to the concatenated, summed result of the
+        ``compare_feedforward`` network, and its output is used as the entailment class logits.
+    premise_encoder : ``Seq2SeqEncoder``, optional (default=``None``)
```

Comment:
What is the motivation for adding this in? Is there some extension paper which added encoders or something? Seems to unnecessarily clutter the implementation and it isn't in the paper.